velox/dwio/parquet/reader/RleBooleanDecoder.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/dwio/parquet/common/BitStreamUtilsInternal.h"
+#include "velox/dwio/parquet/reader/RleBpDecoder.h"
+
+namespace facebook::velox::parquet {
+
+class RleBooleanDecoder : public RleBpDecoder {
+ public:
+  static constexpr int32_t kLengthOffset = 4;
+  RleBooleanDecoder(const char* start, const char* end, int32_t len)
+      : RleBpDecoder{start + kLengthOffset, end, 1} {
+    if (len < kLengthOffset) {
+      VELOX_FAIL("Received invalid length : {} (corrupt data page?)", len);
+    }
+    numBytes_ =
+        ::arrow::bit_util::FromLittleEndian(::arrow::util::SafeLoadAs<uint32_t>(
+            reinterpret_cast<const uint8_t*>(start)));
+    if (numBytes_ > static_cast<uint32_t>(len - kLengthOffset)) {
+      VELOX_FAIL(
+          "Received invalid number of bytes : " + std::to_string(numBytes_) +
```

Comment:
VELOX_FAIL("Received invalid number of bytes: {} (corrupt data page?)"), numBytes_)